source/docs/ondemand_rds_vhds.md

```diff
@@ -0,0 +1,161 @@
+
+## Unsubscribing from Virtual Hosts
+Virtual hosts can be unsubscribed to via a [DeltaDiscoveryRequest](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) with their route config names + domains provided in the resource_names_unsubscribe field. Envoy will remove any route config names + domains that it finds in the [DeltaDiscoveryRequest](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) removed_resources field. 
+
+## Subscribing to Virtual Hosts
+Envoy will send a [DeltaDiscoveryRequest](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) with the resource_names_subscribe field populated with the route config names + domains of each of the resources that it would like to subscribe to. Each of the virtual hosts contained in the [DeltaDiscoveryRequest's](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L137) resources field will be added to the route configuration maintained by Envoy. If Envoy's route configuration already contains a given virtual host, it will be overwritten by data received in the [DeltaDiscoveryResponse](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L193), but only if the updated virtual host is different from its current state. During spontaneous updates configuration server will only send updates for virtual hosts that Envoy is aware of -- the configuration server needs to keep track of virtual hosts known to Envoy.
+
+## Updates of non-virtual host fields
+It might be required to update non-virtual host fields of route configuration while keeping virtual hosts as is. This would probably happen during spontaneous updates only. In such a case a [DeltaDiscoveryResponse](https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/discovery.proto#L193) containing a route configuration with an empty virtual hosts list can be used.
+
+## Compatibility with Scoped RDS
+Both approaches appear to be compatible with scoped RDS proposal: route_configuration name can still be used for vhost matching, but with scoped RDS configured it would point to a scoped route configuration.
+
+Pilot will also support the ability of Envoy to tell it when a resource hasn’t been used and is safe to stop monitoring. The resources that can be removed include the base resources that Pilot initially sent Envoy. 
+
+Any future updates to this resource will be detected by Pilot as part of the normal xDS flow and sent along with normal cluster updates. Istio will continue to use plain xDS for CDS, LDS, EDS, and the other protocols. 
+
+## Full flow example:
+
+![Full flow diagram](delta_rds_full_flow.png)
+
+## Example pilot-generated RDS:
+```json 
+{
```

Comment:
Can you rewrite as YAML? This should be more human readable.

Reply:
Done. Thanks